### DISCUSSION TITLE
Disable Slack Auth but keep integration

### DISCUSSION THREAD

**Discussion by shrinathaithal** [outline/outline#5424] [Self Hosting] [answered]
Hello folks, 

First of all, thanks for the helpful work, absolutely overjoyed to have set this up on AWS using ECS. What simple stuff you folks have built!

Now, I have enabled Google login, and I want only that to be the way SSO login way.   
I want to use the Slack app for /outline command purposes, but do not want to show that in login screen. 

How do I do that?

**Comment by tommoor** [comment]
You will need to look in the database under `authentication_providers` and change the `enabled` column to `FALSE`.

**Reply by shrinathaithal** [reply]
Sorry, but that table in db contains only one column - Google. 
This is probably because it is a self hosted community edition which doesn't allow more than 1 auth types?

**Reply by tommoor** [reply]
In that case, Slack should not be shown on the login screen

**Reply by shrinathaithal** [reply]
🤔 But it is! There's literally single row in that table and I see 2 logins. 

Sorry to be shoving more data into this, but here are config/env settings may cause this - 

I run this on AWS ECS, with Postgres RDS for DB. 
Version 0.69.2.

Here are the Slack related things I've set for env. 
```
SLACK_APP_ID=<hidden>
SLACK_CLIENT_ID=<hidden>
SLACK_CLIENT_SECRET=<hidden>		
SLACK_MESSAGE_ACTIONS=true
SLACK_VERIFICATION_TOKEN=<hidden>
```
Now, it shows the following screen on trying to auth with Slack. 

<img width="382" alt="Screenshot 2023-06-09 at 6 55 23 AM" src="https://github.com/outline/outline/assets/5593857/d9cd6d2c-b347-4839-a203-01c495c694ba">

And this is printed in logs. 

```JSON
{
  "error": "The maximum number of teams has been reached",
  "level": "error",
  "message": "Error during authentication",
  "stack": "BadRequestError: The maximum number of teams has been reached\n at MaximumTeamsError (/opt/outline/build/server/errors.js:125:34)\n at teamProvisioner (/opt/outline/build/server/commands/teamProvisioner.js:70:43)\n at process.processTicksAndRejections (node:internal/process/task_queues:95:5)"
}
```

Did I do something wrong?

**Reply by tommoor** [reply]
I see, yes – the code changed here recently. I think you can remove all slack variables except `SLACK_VERIFICATION_TOKEN` and the slash command will continue to work.

**Reply by shrinathaithal** [reply]
Did that. Immediately got an error while starting it - 
```
{"level":"warn","message":"- SLACK_VERIFICATION_TOKEN cannot be used without SLACK_CLIENT_ID."}
```

Added that. Then got this - 
```
{"level":"warn","message":"- SLACK_CLIENT_ID cannot be used without SLACK_CLIENT_SECRET."}
```

Added that. Now running fine, but shows Slack as an option to login with!

**Reply by tommoor** [reply]
Lol ok, this will require a new release to fix then

**Reply by shrinathaithal** [reply, marked as answer]
Awesome! Thanks Tom.

I am not ready to give up yet. Given that I can access database, I am gonna do one more somewhat silly attempt.
I could create another row in `authentication_providers` table, fill it with junk data except `name` and `teamID`, and set `enabled = false`. 
Will let you know if it works.

**Reply by shrinathaithal** [reply]
Ah! That worked. Amazing. 
Now I can also see in Workspace > Security I have 3 sign in options, slack is disabled. 

This is amazing, thanks Tom. 
This is a really good piece of software you folks have given us here. Small things apart, it is really a no bloatware no frills setup. Was relatively easy to self host as well. ❤️

**Reply by tommoor** [reply]
Great, removed the requirement for `SLACK_ID` in this commit too 

3f7e66980b31e955e3aeb6ea31c30c96e05b8eda

**Reply by shrinathaithal** [reply]
That was fast!! 🔥 
Thanks Tom.
